Tests/iaas/entropy/entropy-check.py

```diff
@@ -0,0 +1,377 @@
+"""
+from collections import Counter
+import getopt
+import logging
+import os
+import re
+import sys
+import tempfile
+import time
+import warnings
+
+import fabric
+import invoke
+import openstack
+import openstack.cloud
+
+
+logger = logging.getLogger(__name__)
+
+# use telling names here, maybe even reuse the security group in case of a leak?
+SERVER_NAME = "scs-0101-server"
+SECURITY_GROUP_NAME = "scs-0101-group"
+KEYPAIR_NAME = "scs-0101-keypair"
+
+IMAGE_ATTRIBUTES = {
+    # https://docs.openstack.org/glance/latest/admin/useful-image-properties.html#image-property-keys-and-values
+    # type: str
+    "hw_rng_model": "virtio",
+}
+FLAVOR_ATTRIBUTES = {
+    # https://docs.openstack.org/nova/latest/configuration/extra-specs.html#hw-rng
+    # type: bool
+    "hw_rng:allowed": True,
+}
+FLAVOR_OPTIONAL = ("hw_rng:rate_bytes", "hw_rng:rate_period")
+
+
+def print_usage(file=sys.stderr):
+    """Help output"""
+    print("""Usage: entropy-check.py [options]
+This tool checks the requested images and flavors according to the SCS Standard 0101 "Entropy".
+Options:
+ [-c/--os-cloud OS_CLOUD] sets cloud environment (default from OS_CLOUD env)
+ [-d/--debug] enables DEBUG logging channel
+ [-i/--images IMAGE_LIST] sets images to be tested, separated by comma.
+""", end='', file=file)
+
+
+def check_image_attributes(images, attributes=IMAGE_ATTRIBUTES):
+    for image in images:
+        wrong = [f"{key}={value}" for key, value in attributes.items() if image.get(key) != value]
+        if wrong:
+            logger.info(f"Image '{image.name}' missing recommended attributes: {', '.join(wrong)}")
+
+
+def check_flavor_attributes(flavors, attributes=FLAVOR_ATTRIBUTES, optional=FLAVOR_OPTIONAL):
+    for flavor in flavors:
+        extra_specs = flavor['extra_specs']
+        wrong = [f"{key}={value}" for key, value in attributes.items() if extra_specs.get(key) != value]
+        miss_opt = [key for key in optional if extra_specs.get(key) is None]
+        if wrong:
+            message = f"Flavor '{flavor.name}' missing recommended attributes: {', '.join(wrong)}"
+            # only report missing optional attributes if recommended are missing as well
+            # reasoning here is that these optional attributes are merely a hint for implementers
+            # and if the recommended attributes are present, we assume that implementers have done their job already
+            if miss_opt:
+                message += f"; additionally, missing optional attributes: {', '.join(miss_opt)}"
+            logger.info(message)
+
+
+def install_test_requirements(fconn):
+    # the following commands seem to be necessary for CentOS 8, but let's not go there
+    # because, frankly, that image is ancient
+    # sudo sed -i -e "s|mirrorlist=|#mirrorlist=|g" /etc/yum.repos.d/CentOS-*
+    # sudo sed -i -e "s|#baseurl=http://mirror.centos.org|baseurl=http://vault.centos.org|g" /etc/yum.repos.d/CentOS-*
```

Comment:
Supported? What do you mean? Supported by whom to do what? CentOS 8 is one of the recommended images in the Image Metadata Standard. I added a point to tomorrow's agenda of Team IaaS to discuss this matter, because CentOS 8 has been EOL since 2021 or so. At any rate, I don't get why that comment poses a problem.

Reply:
> we should decide - I don't know at least - if EL8 is supported or not.

To clarify, I think I now understand what you mean, and I don't disagree -- what @cah-hbaum and I are actually seeking out from this reviewing process is precisely the opinions of the reviewers on whether CentOS 8 is relevant here or not... Would you provide your opinion?

Reply:
sure! to make it short: given that most EL8 releases are EOL, I would not support it, as it can't be easily tested - as is evident in this PR - by only using freely available releases.